src/poll.rs

```diff
@@ -56,14 +90,72 @@ fn emit_event(event_handler: &Mutex<dyn EventHandler>, res: Result<Event>) {
     }
 }
 
+impl PathData {
+    pub fn collect<BH: BuildHasher>(
+        path: &Path,
+        metadata: &Metadata,
+        build_hasher: Option<&BH>,
+        last_check: Instant,
+    ) -> Self {
+        let mtime = FileTime::from_last_modification_time(metadata).seconds();
+        let hash = metadata
+            .is_file()
+            .then(|| build_hasher.and_then(|bh| Self::hash_file(path, bh).ok()))
+            .flatten();
+        Self {
+            mtime,
+            hash,
+            last_check,
+        }
+    }
+
+    fn hash_file<P: AsRef<Path>, BH: BuildHasher>(path: P, build_hasher: &BH) -> io::Result<u64> {
+        let mut hasher = build_hasher.build_hasher();
+        let mut file = fs::File::open(path)?;
+        let mut buf = [0; 512];
+        while file.read(&mut buf)? > 0 {
+            hasher.write(&buf);
```

Comment:
Also this would be faulty, because we'd have to use hash.write(&buf[..length_read_last])

Reply:
Oops good catch.  I'm new to rust and don't have the basics on auto pilot yet :)

Re io::copy, i looked into this but I don't think Hasher implements Write so it won't work?

Reply:
Ah yeah you're right..

Reply:
FWIW, the Digest crate (that has MD5, SHA1/2/3, etc) _does_ implement the Write attribute, so it's much more convenient to use like this.  I am guessing Rust devs just thought of Hasher as used for stuff like HashMaps and not general purpose file hashing like the other more sophisticated digests.  I could actually switch to using them, but I can't see the point given the tests I linked elsewhere showing that the std Rust Hasher doesn't collide once with a huge 35M word list...

Reply:
The default hasher is DoS resistant, so it's not the fastest. But since most people also don't change that for HashMap we should be fine (and secure).